### ISSUE TITLE
Function parameter syntax using parentheses

### ISSUE THREAD

**Issue by vwkd** [nushell/nushell#4941]
### Related problem

For the most part Nu sticks to C-style syntax. One glaring exception is the choice of square brackets for function parameters what in most common languages instead are parentheses.

This seems like an obvious stumbling block for most programmers coming to Nu.

### Describe the solution you'd like

Parentheses for function parameters.

```
def greet (name) {
  echo "hello" $name
}

greet "world"
```

**Comment by jntrnr** [comment]
This doesn't feel like a good fit for Nushell - this is because the way you call a function and the way you define a function has a connection. In languages like JS/C/Rust, you call the function using `()` and define functions using `()`. There's a natural connection.

When we talked about the design for custom commands, we though about using `()` as well, but the problem is that it gives the wrong mental model for how to call the command you just created. Instead, we used a different syntax that doesn't have that association.

Some languages that don't use `()` when calling functions define the parameter list without any containing delimiter (Haskell, OCaml, etc). This also didn't feel like a good fit for Nushell because it has both named and positional parameters.

In short, we compromised and picked `[]` as a unique containing syntax to hold all of these parameter types in one place, but without confusing the user in how this thing would later be called.

**Comment by hustcer** [comment]
This is a breaking change that is not appropriate to introduce at this stage, I will close this issue, Thanks for your suggestion

**Comment by hustcer** [comment]
It's supported since v0.68 by https://github.com/nushell/nushell/pull/6444
